pkg/api/validation/validation.go

```diff
@@ -1487,19 +1487,21 @@ func ValidateResourceRequirements(requirements *api.ResourceRequirements) errs.V
 	allErrs := errs.ValidationErrorList{}
 	for resourceName, quantity := range requirements.Limits {
 		// Validate resource name.
-		errs := validateResourceName(resourceName.String(), fmt.Sprintf("resources.limits[%s]", resourceName))
+		allErrs = append(allErrs, validateResourceName(resourceName.String(), fmt.Sprintf("resources.limits[%s]", resourceName))...)
 		if api.IsStandardResourceName(resourceName.String()) {
-			errs = append(errs, validateBasicResource(quantity).Prefix(fmt.Sprintf("Resource %s: ", resourceName))...)
+			allErrs = append(allErrs, validateBasicResource(quantity).Prefix(fmt.Sprintf("Resource %s: ", resourceName))...)
+		}
+		requestQuantity, exists := requirements.Requests[resourceName]
+		if exists && quantity.MilliValue() < requestQuantity.MilliValue() {
```

Comment:
I also think you need to make the quantity version different if the resource is cpu or memory.

I think for memory, you want to do ```quantity.Value() < requestQuantity.Value()``` and for cpu, you want to use ```MilliValue()```.   Always using MilliValue can cause an overflow.

https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/api/resource/quantity.go#L352